### ISSUE TITLE
FYI: under x2go experiences 1-2 min import time

### ISSUE THREAD

**Issue by yarikoptic** [jaraco/keyring#420]
- Initially reported against datalad: https://github.com/datalad/datalad/issues/4085
- I never used x2go, installed and reproduced:
```shell
$> time python -c 'import keyring'
Gkr-Message: 22:13:50.844: secret service operation failed: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
python -c 'import keyring'  0.11s user 0.04s system 0% cpu 50.197 total
```
- keyring installed via debian package 17.1.1-1
- Relates to #403 (on import times) -- IMHO keyring must not load any backend merely upon import. Should load it (ideally the one "best" for the system or specified, not all) only whenever its API started to be used and require connection.
- decided to file an issue just to let you know and didn't want to "pollute" #403, so feel free to close unless you want to convert it to some action (e.g. "add a reasonable timeout for allowing backend to load")

**Comment by takluyver** [comment]
This is hopefully improved from version 21.5.0, following PR #474. Now, on import, keyring only checks whether the secrets & kwallet services are available, rather than trying to talk to them. If the code uses keyring, it may still see a delay when it does that, but fingers crossed that's better than a long wait on import.

**Comment by adfnekc** [comment]
see more in [https://github.com/pypa/pip/issues/8485](https://github.com/pypa/pip/issues/8485)

**Comment by jaraco** [comment]
This should be fixed with the latest release due to #480.
